var/spack/repos/builtin/packages/ncview/package.py

```diff
@@ -31,7 +39,12 @@ def configure_args(self):
         config_args.append("--with-udunits2_incdir={}".format(spec["udunits"].prefix.include))
         config_args.append("--with-udunits2_libdir={}".format(spec["udunits"].prefix.lib))
 
-        if spec.satisfies("^netcdf-c+mpi"):
-            config_args.append("CC={}".format(spec["mpi"].mpicc))
+        #if spec.satisfies("^netcdf-c+mpi"):
+        #    config_args.append("CC={}".format(spec["mpi"].mpicc))
+
+        #nc_config = which("nc-config")
+        #cc = nc_config("--cc", "2>&1")
+        cc = subprocess.check_output(['nc-config', '--cc']).decode().rstrip('\n')
+        config_args.append("CC={}".format(cc))
```

Comment:
What was going wrong with the previous configuration? Also, is there ever conceivably a time where the compiler there wouldn't be available, like maybe if netcdf-c was external?

Reply:
THAT is a complicated story. Systems that are Cray but no longer recognized as such (now recognized as "ordinary" linux systems, a feature of the recent spack develop branch) but still using the Cray compiler wrappers had a mismatch between the `netcdf-c` and the `ncview` compiler with the old config. The compiler is `cc` for `netcdf-c` (because of the wrapper), but `spec["mpi"].mpicc` pointed to `/path/to/mpicc`. This does't even exist on systems like Narwhal. There's been a 83-messages long back and forth between the spack developers and me about the best way, they recommended updating the entire Narwhal site config to newer versions of essentially the entire Cray stack, but that broke a ton of other stuff. I also tried to remove the compiler check from `configure` (as you can see from the comments that still need to be removed), but that triggered the `autoconf` reconfigure and that didn't work on some of the other Cray systems. Grrrrr ... This was the best solution I could come up with to match the `netcdf-c` compiler.